coderd/database/migrations/000274_check_tags.up.sql

```diff
@@ -0,0 +1,16 @@
+-- We need this as a type alias to ensure that we can override it in sqlc.
+-- We need to override it in sqlc so that it can be recognised as a StringMap.
+-- Without this zero values for other inferred types cause json syntax errors.
+CREATE DOMAIN tags AS jsonb;
+
+CREATE OR REPLACE FUNCTION tags_compatible(subset_tags tags, superset_tags tags)
+RETURNS boolean as $$
+BEGIN
+	RETURN CASE
+		-- Special case for untagged provisioners
+		WHEN subset_tags :: jsonb = '{"scope": "organization", "owner": ""}' :: jsonb
+		THEN subset_tags = superset_tags
+		ELSE subset_tags :: jsonb <@ superset_tags :: jsonb
```

Comment:
I imagine we don't need the type casts for the named vars here since that's already covered by args (as well as in the WHEN case)?

Reply:
I find that sqlc generally does better when you explicitly add type casts, maybe this is what's happening?

Reply:
In this function they are likely optional, but I wholeheartedly agree that sqlc is much, much happier if you give the type inference some hints. If there's a string motivation to remove, I'll do so. Otherwise, I'm happy to leave this as is.

Reply:
My motivation to remove is that it improves clarity and makes the query easier to read. No point in clogging up the query with added syntax unless it's to help sqlc.

Reply:
Removing

Reply:
Done